### ISSUE TITLE
Ant Menu item doesn't support Select component inside

### ISSUE THREAD

**Issue by useless-stuff** [ant-design/ant-design#7679]
### Version
2.13.2

### Environment
Mac OS, latest Chrome/Firefox/Opera, latest React 

### Reproduction link
[https://codepen.io/rieke/pen/OxNBLx?editors=0010](https://codepen.io/rieke/pen/OxNBLx?editors=0010)

### Steps to reproduce
Attaching a Select component in the menu Element

### What is expected?
When I click on the Select Option I'm expecting that the menu will not closed

### What is actually happening?
When you click on the Select component the Menu disappear

---
I don't know if it could be considered a bug or an improvment request

**Comment by benjycui** [comment]
Yep, but why do you need to place a Select in Menu??

**Comment by useless-stuff** [comment]
![screen shot 2017-09-21 at 12 08 39](https://user-images.githubusercontent.com/1117752/30693037-c2d2ab80-9ec5-11e7-9418-b01d9836ff14.png)
I have a case where I need to put a select component in the menu but there isn't a way to catch that event even if using the container option

**Comment by useless-stuff** [comment]
![screen shot 2017-09-21 at 12 14 55](https://user-images.githubusercontent.com/1117752/30693241-8e6bf6ac-9ec6-11e7-88da-88e6941c106a.png)
This is the entire menu

**Comment by afc163** [comment]
It is a huge different menu from antd's default menu. I suggest you try a customized Popover for select component.

**Comment by useless-stuff** [comment]
@afc163 Thanks for the answer, I'm trying to use the popover and still have the same problem but I'm still working on looking for a solution :)
I'll let you know ( just in case someone else is interested to know ) if I'll able to do that through the popover.

Thank you so much!

**Comment by Waserman** [comment]
@useless-stuff did you find any solution for this behaviour? Im trying to place antd Select component inside a Popover, and whenever I chose one of it's options, the Popover close. 

I really don't think that's a matter of misuse, but a buggy behaviour.

**Comment by useless-stuff** [comment]
To be honest we moved away from the ant menu ( we built a new one from
scratch because of lots of custom components over the original ).
I remember that we got the same issue about the select component and we
solved it using react-click-outside and attaching the component to a
specific container through getPopupContainer.

https://www.npmjs.com/package/react-click-outside

On 25 Dec 2017 10:03, "Avi Waserman" <notifications@github.com> wrote:

@useless-stuff <https://github.com/useless-stuff> did you find any solution
for this behaviour? Im trying to place antd Select component inside a
Popover, and whenever I chose one of it's options, the Popover close.

I really don't think that's a matter of misuse, but a buggy behaviour.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/ant-design/ant-design/issues/7679#issuecomment-353851383>,
or mute the thread
